### PR TITLE
Hotfix: OutOfMemory crashes

### LINKED ISSUE #2449: InflateException: Binary XML file line #23: Error inflating class <unknown>
Sentry Issue: [WOOCOMMERCE-ANDROID-JV](https://sentry.io/organizations/a8c/issues/1655429167/?referrer=github_integration)

```
OutOfMemoryError: Failed to allocate a 880992 byte allocation with 741112 free bytes and 723KB until OOM
    at com.woocommerce.android.ui.main.MainActivity.getTheme(MainActivity.kt:127)
    at com.woocommerce.android.ui.orders.OrderDetailCustomerInfoView.<init>(OrderDetailCustomerInfoView.kt:28)
    at com.woocommerce.android.ui.orders.OrderDetailCustomerInfoView.<init>(OrderDetailCustomerInfoView.kt:25)
    at com.woocommerce.android.ui.orders.OrderDetailCustomerInfoView.<init>(OrderDetailCustomerInfoView.kt)
...
(39 additional frame(s) were not displayed)

InvocationTargetException: None
    at java.lang.reflect.Constructor.newInstance(Constructor.java)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
    at android.view.LayoutInflater.createView(LayoutInflater.java:614)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:750)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:813)
...
(20 additional frame(s) were not displayed)

InflateException: Binary XML file line #23: Error inflating class <unknown>
    at com.woocommerce.android.ui.orders.OrderFulfillmentFragment.onCreateView(OrderFulfillmentFragment.kt:49)
...
(18 additional frame(s) were not displayed)

Binary XML file line #23: Error inflating class <unknown>
```

### PR DESCRIPTION
**Pull request by AmandaRiu**
Fixes #2449 and #2417 by only force applying the `DayNight` theme in `MainActivity` once per session.

**Note:** Sentry has a bunch of `OutOfMemory` crash events logged with stacktraces showing the crash happening in different areas of the app, but you'll know if the crash is related to this fix if you see `MainActivity.getTheme(...)` in the stack dump. 

**This is being submitted as a hotfix because this bug essentially breaks the app for users running Android Lollipop, as well as crashes and poor performance on newer versions of Android.** 

The actual cause of the crash in the ticket is an `OutOfMemoryError` caused by the new logic that applies the `DayNight` theme in the `MainActivity` to allow for displaying a splash screen while the main activity loads:

```
override fun getTheme(): Theme {
    return super.getTheme().also { it.applyStyle(R.style.Theme_Woo_DayNight, true) }
}
```

Every time a new view is loaded the above `getTheme()` method is called and this was leading to a TON of overhead as the current theme was completely overwritten and applied over and over again for _every view rendered_. This led to eventual OOM crashes on Android 5.1 and heavy GC and possible crashes in all versions of Android.

This crash can be reproduced on an android emulator running API 5.1 w/ 512 MB of RAM. Load the previous version of the app and then:
1. Navigate to the orders list
2. Click on an order to view the detail
3. Click back to return to the orders list
4. Repeat steps 2 and 3 until the app crashes (took me about 15 orders to crash)

The fix was to ensure this theme is only applied once per session: 
```
if (!isMainThemeApplied) {
    it.applyStyle(R.style.Theme_Woo_DayNight, true)
    isMainThemeApplied = true
} }
```

## To Test
- Try the same steps outlined above to verify the issue is fixed.
- Also verify the correct light/dark theme loads throughout the app across configuration changes (API 21 and API 28, 29)

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

cc: @oguzkocer

### CODE REVIEW

**Comment by nbradbury** [comment]
@AmandaRiu Would it be simpler to just set the theme in the activity's `onCreate()`, like this:

```
override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)

        setTheme(R.style.Theme_Woo_DayNight)
        setContentView(R.layout.activity_main)
```

**Comment by AmandaRiu** [comment]
@nbradbury 

> Would it be simpler to just set the theme in the activity's onCreate()

I'm pretty sure I tried that first and it didn't work across all versions and caused UI flicker. Let me try again just in case.

**Comment by AmandaRiu** [comment]
@nbradbury Well, I'm not sure what nonsense I was running into before, but I tested on API 21, 23, 28 and 29 - they all work fine. Thanks for the suggestion. I like this implementation much better 👍 Ready for another round.

**Comment by AmandaRiu** [comment]
Note: While testing the final hotfix build I ran into the original reason why I used `getTheme()` instead of just setting in `onCreate()`. Will create a new PR with a different fix. 

For historical purposes, here's how I got this to fail (Android API 29):
1. log into the app
2. click on the orders tab
3. while the orders are loading, rotate the device - theme is reverted back to Splash
